src/analysis/inner_product_space/reproducing_kernel.lean

```diff
@@ -0,0 +1,256 @@
+
+end complete_space
+
+/-!
+## Scalar functionals
+
+In this section, we focus on the case `V = 𝕜`.
+-/
+
+section scalar
+
+omit hHrkhs
+
+section defs
+
+variables (𝕜 X)
+variables (H₁ : Type*) [inner_product_space 𝕜 H₁] [complete_space H₁] [hH₁rkhs : rkhs 𝕜 X 𝕜 H₁]
+
+include hH₁rkhs
+
+/--
+Evaluation at a point `x`, represented as a point `eval' x` in the RKHS `H₁` by the Riesz
+representation theorem.
+-/
+def eval' (x : X) : H₁ := (to_dual 𝕜 H₁).symm (eval 𝕜 𝕜 H₁ x)
+
+/--
+The kernel of an RKHS, represented as a scalar value.
+-/
+def scalar_kernel (x y : X) : 𝕜 := (to_dual 𝕜 𝕜).symm (kernel 𝕜 𝕜 H₁ x y)
+
+/--
+The span of all points of the form `eval' x` as a subspace of `H₁`.
+-/
+def eval'_span : submodule 𝕜 H₁ := submodule.span 𝕜 $ set.range (eval' 𝕜 X H₁)
+
+end defs
+
+variables {H₁ : Type*} [inner_product_space 𝕜 H₁] [complete_space H₁] [hH₁rkhs : rkhs 𝕜 X 𝕜 H₁]
+include hH₁rkhs
+
+lemma eval'_def (x : X) : eval' 𝕜 X H₁ x = (to_dual 𝕜 H₁).symm (eval 𝕜 𝕜 H₁ x) := rfl
+
+lemma inner_eval' (f : H₁) (x : X) : inner (eval' 𝕜 X H₁ x) f = f x :=
+by rw [eval'_def, to_dual_symm_apply, eval_apply]
+
+lemma eval'_eq_eval_adjoint (x : X) : eval' 𝕜 X H₁ x = (eval 𝕜 𝕜 H₁ x)† 1 :=
+begin
+  apply (to_dual 𝕜 H₁).injective,
+  ext f,
+  simp [eval'_def],
+end
+
+
+lemma scalar_kernel_def (x y : X) :
+  scalar_kernel 𝕜 X H₁ x y = (to_dual 𝕜 𝕜).symm (kernel 𝕜 𝕜 H₁ x y) := rfl
+
+lemma scalar_kernel_eq_inner (x y : X) :
+  scalar_kernel 𝕜 X H₁ x y = inner (eval' 𝕜 X H₁ y) (eval' 𝕜 X H₁ x) :=
+begin
+  apply (to_dual 𝕜 𝕜).injective,
+  rw [scalar_kernel_def, linear_isometry_equiv.apply_symm_apply, kernel_def],
+  ext,
+  rw [continuous_linear_map.comp_apply, ←eval'_eq_eval_adjoint],
+  simp only [eval_apply, to_dual_apply, is_R_or_C.inner_apply, inner_conj_sym, mul_one],
+  rw inner_eval',
+end
+
+lemma kernel_conj_symm {x y : X} :
+  scalar_kernel 𝕜 X H₁ x y = star_ring_end 𝕜 (scalar_kernel 𝕜 X H₁ y x) :=
+by rw [scalar_kernel_eq_inner, scalar_kernel_eq_inner, inner_conj_sym]
+
+lemma norm_eval'_eq_kernel {x : X} :
+  ∥ eval' 𝕜 X H₁ x ∥^2 = is_R_or_C.re (scalar_kernel 𝕜 X H₁ x x) :=
+by rw [←inner_self_eq_norm_sq, scalar_kernel_eq_inner]
+
+lemma norm_eval_eq_kernel {x : X} :
+  ∥ eval 𝕜 𝕜 H₁ x ∥^2 = is_R_or_C.re (scalar_kernel 𝕜 X H₁ x x) :=
+by rw [←norm_eval'_eq_kernel, eval', linear_isometry_equiv.norm_map]
+
+/--
+The span of the `eval'`s is dense in H
+-/
+lemma eval'_span_topological_closure_eq_top :
+  (eval'_span 𝕜 X H₁).topological_closure = ⊤ :=
+begin
+  rw [submodule.topological_closure_eq_top_iff, submodule.eq_bot_iff],
+  intros f hf,
+  apply fun_like.coe_injective,
+  funext x,
+  specialize hf (eval' 𝕜 X H₁ x) _,
+  rw [eval'_span],
+  apply submodule.subset_span,
+  use x,
+  rw inner_eval' at hf,
+  rw [hf, zero_apply],
+end
+
+/--
+If `f n` tendsto `F` in `H`, then `f n` tendsto `F` pointwise.
+-/
+lemma tendsto_nhds_apply (f : ℕ → H₁) (F : H₁) (hf : tendsto f at_top (𝓝 F))
+  (x : X) : tendsto (λ n, f n x) at_top (𝓝 (F x)) :=
```

Comment:
There's probably a strengthening to filters other than `at_top` and `ℕ` indexed functions?